examples/3d_kymograph.py

```diff
@@ -0,0 +1,148 @@
+except:
+    print("Could not import BlitzGateway which is")
+    print("required to download the sample datasets.")
+    print("Please install omero-py:")
+    print("https://pypi.org/project/omero-py/")
+    exit(-1)
+
+
+def IDR_fetch_image(image_id: int, progressbar: bool = True) -> np.ndarray:
+    """
+    Download the image with id image_id from the IDR
+
+    Will fetch all image planes corresponding to separate
+    timepoints/channels/z-slices and return a numpy
+    array with dimension order (t,z,y,x,c)
+
+    Displaying download progress can be disabled by passing
+    False to progressbar.
+    """
+
+    conn = BlitzGateway(
+        host="ws://idr.openmicroscopy.org/omero-ws",
+        username="public",
+        passwd="public",
+        secure=True,
+    )
+    conn.connect()
+    conn.c.enableKeepAlive(60)
+
+    idr_img = conn.getObject("Image", image_id)
+    idr_pixels = idr_img.getPrimaryPixels()
+
+    _ = idr_img
+    nt, nz, ny, nx, nc = (
+        _.getSizeT(),
+        _.getSizeZ(),
+        _.getSizeY(),
+        _.getSizeX(),
+        _.getSizeC(),
+    )
+
+    plane_indices = list(product(range(nz), range(nc), range(nt)))
+    idr_plane_iterator = idr_pixels.getPlanes(plane_indices)
+
+    if progressbar:
+        idr_plane_iterator = tqdm(idr_plane_iterator, total=len(plane_indices))
+
+    _tmp = np.asarray(list(idr_plane_iterator))
+    _tmp = _tmp.reshape((nz, nc, nt, ny, nx))
+    # the following line reorders the axes (no summing, despite the name)
+    return np.einsum("jmikl", _tmp)
+
+
+description = """
+3D-Kymographs in Napari
+=======================
+
+About
+=====
+This example demonstrates that the volume rendering capabilities of napari
+can also be used to render 2d timelapse acquisitions as kymographs. 
+Kymographs, also called space-time images, are a powerful tool to visualize
+the dynamics of processes.  
+The most common way to visualize kymographs is to pick a single line through
+a 2D image and visualize the time domain along a second axes. 
+Napari is not limited to 2D visualization an by harnessing its volume
+volume rendering capabilities, we can create a 3D kymograph, 
+a powerful visualization that provides an overview of the complete 
+spatial and temporal data from a single view.
+
+Using napari's grid mode we can juxtapose multiple such 3D kymographs to
+highlight the differences in cell dynamics under different siRNA treatments. 
+
+The selected samples are from the Mitocheck screen and demonstrate siRNA 
+knockdowns of several genes. 
+The date is timelapse fluorescence microscopy of HeLa cells, with GFP-
+tagged histone revealing the chromosomes.
+
+In the juxtaposed kymographs the reduced branching for the mitotitic 
+phenotypes caused by INCENP, AURKB and KIF11 knockdown compared to 
+TMPRSS11A knockdown is immediately obvious.
+
+Data Source
+===========
+The samples to demonstrate this is downloaded from IDR:
+https://idr.openmicroscopy.org/webclient/?show=screen-1302
+
```

Comment:
It would be good to add how large the dataset is (in GB or similar), and roughly how long you should expect it to take to download (just an estimate).

Reply:
Agreed. Takes about 2min iirc.

Reply:
Good suggestion. I will provide a warning about the download size.
Do you think I should ask for keypress? I.e. "Do you want to go ahead with the download? [Enter to confirm]"

I can't really estimate the download speed as that will depend on the user's internet connection and whatever bandwidth limiting IDR may have in place.
I think the tqdm progress bars should give a reasonable estimate though.

Reply:
A keypress confirmation is a good idea, I like that.

> I think the tqdm progress bars should give a reasonable estimate though.

Ok, no need to add a comment about the expected time. Users can always stop the script if the download speed looks super slow for some reason.